shared/actions/teams/index.js

```diff
@@ -0,0 +1,72 @@
+// @flow
+import * as I from 'immutable'
+import * as Constants from '../../constants/teams'
+import * as ChatConstants from '../../constants/chat'
+import * as ChatTypes from '../../constants/types/flow-types-chat'
+import * as Saga from '../../util/saga'
+import * as Creators from './creators'
+import {replaceEntity} from '../entities'
+import {call, put, select} from 'redux-saga/effects'
+import {usernameSelector} from '../../constants/selectors'
+
+import type {SagaGenerator} from '../../constants/types/saga'
+
+const _getChannels = function*(action: Constants.GetChannels): SagaGenerator<any, any> {
+  const results = yield call(ChatTypes.localGetTLFConversationsLocalRpcPromise, {
+    param: {
+      includeAuxiliaryInfo: false,
+      membersType: ChatTypes.CommonConversationMembersType.team,
+      tlfName: action.payload.teamname,
+      topicType: ChatTypes.CommonTopicType.chat,
+    },
+  })
+
+  const entityMap = results.convs.reduce((map, conv) => {
+    const name = conv.info.topicName
+    map[name] = Constants.ChannelRecord({
+      conversationIDKey: ChatConstants.conversationIDToKey(conv.info.id),
+      participants: I.Set([].concat(conv.info.readerNames, conv.info.writerNames)),
+    })
+    return map
+  }, {})
+
+  yield put(replaceEntity(['teams', action.payload.teamname, 'channels'], entityMap))
```

Comment:
Not a big fan of this being so deep. You have to know the teamname and the topicname to get this.

What if we had this structure:
```
{teams:
  {convIDToParticipants: [ConversationIDKey: key]: Particpants
   convIDToTeamName: ...
   convIDToTopicName: ...
   teamNameToConvIDs: ...
 }
}
```
We can discuss